plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/linked-product-control.tsx

```diff
@@ -40,6 +42,12 @@ const ProductButton: React.FC< {
 		return <Spinner />;
 	}
 
+	const showPlaceholder = ! product;
```

Comment:
Can we also check whether the image source is `undefined` to show the placeholder? Otherwise,  when the product doesn't have an image will look like this

![Screenshot 2024-09-17 at 10 39 03 AM](https://github.com/user-attachments/assets/a18fbca6-aea8-4c53-b20b-7947ddb62c41)

Reply:
If we add a product here that we send later to the trash can, should we continue showing it?

Reply:
Oh, nice catch! <3

Reply:
> If we add a product here that we send later to the trash can, should we continue showing it?

Good thinking! We have it covered in the upper scope. So basically if the product is not available, merchant won't even get to this state but will be prompted to choose another product earlier.

It was Implemented in this PR https://github.com/woocommerce/woocommerce/pull/51114 and here's how it looks like, the promp shows up instead of the Product Collection block:

![image](https://github.com/user-attachments/assets/c9937b3a-4bb7-4132-94c1-1b4f4e8414a6)